### PR TITLE
Fix doc warnings.

### LINKED ISSUE #1254: Fix sphinx warnings
Sphinx currently raises a large number of warnings when docs are built. So that doc builds are easier to debug, and errors are more likely to be caught, fix or filter these warnings.

### PR DESCRIPTION
**Pull request by jesnie**
Fix doc warnings. Fixes #1254

Fix all warnings generated by Sphinx. This doesn't mean our documentation is far perfect - far from it.
But at least some classes of problems have been fixed.

For maths it has been a bit of a mix whether I've formatted it as a code example:

```
Example::

    math goes here
```

or as LaTeX:

```
Example:

... math::

    math goes here
```

obviously LaTeX is perttier, but it's also much more work. I figure we'll need to go back and look over all of it again anyway.

### CODE REVIEW

File: doc/sphinx/intro.rst [hunk 1]
```diff
@@ -77,7 +77,7 @@ mathematical background and the resulting software design is described in :cite:
 GPLVM
 """""
 For visualisation, the GPLVM :cite:p:`lawrence2003gaussian` and Bayesian GPLVM :cite:p:`titsias2010bayesian` models are implemented
-in GPflow (`notebook <notebooks/basics/GPLVM.html>`_).
+in GPflow (`GPLVM notebook <notebooks/basics/GPLVM.html>`_).
```
Comment (jesnie) [inline doc/sphinx/intro.rst:80]: ... and this was the other `notebook` link.

File: doc/sphinx/notebooks/advanced/natural_gradients.pct.py [hunk 2]
```diff
@@ -52,7 +52,7 @@
 autotune = tf.data.experimental.AUTOTUNE
 
 # %% [markdown]
-# ### VGP is a GPR
+# ## VGP is a GPR
```
Comment (jesnie) [inline doc/sphinx/notebooks/advanced/natural_gradients.pct.py:55]: Sphinx complained something something an inconsistent header in this file, so I redid all of them. :shrug:

File: doc/sphinx/notebooks/theory/cglb.pct.py [hunk 3]
```diff
@@ -28,6 +28,8 @@
 # %matplotlib inline
 
 # %% [markdown]
+# # Conjugate Gradient Lower Bound
```
Comment (jesnie) [inline doc/sphinx/notebooks/theory/cglb.pct.py:31]: Sphinx requires all files to have a header.

File: gpflow/likelihoods/base.py [hunk 4]
```diff
@@ -11,47 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-"""
```
Comment (jesnie) [inline gpflow/likelihoods/base.py:15]: All this text was moved to: `gpflow/likelihoods/__init__.py`.

File: gpflow/likelihoods/__init__.py [hunk 5]
```diff
@@ -11,6 +11,46 @@
...
+Likelihoods are another core component of GPflow. This describes how likely the
+data is under the assumptions made about the underlying latent functions
+p(Y|F). Different likelihoods make different
+assumptions about the distribution of the data, as such different data-types
+(continuous, binary, ordinal, count) are better modelled with different
+likelihood assumptions.
+
+Use of any likelihood other than Gaussian typically introduces the need to use
+an approximation to perform inference, if one isn't already needed. A
+variational inference and MCMC models are included in GPflow and allow
```
Comment (frgsimpson) [inline gpflow/likelihoods/__init__.py:24]:
```suggestion
an approximation to perform inference, if one isn't already needed. 
Variational inference and MCMC models are included in GPflow and allow
```

**Review by frgsimpson — APPROVED** [review]
LGTM!

**Pull request merged**